lib/mergeArrayByName.js

```diff
@@ -0,0 +1,34 @@
+// https://github.com/KyleAMathews/deepmerge#arraymerge
+
+const merge = require('deepmerge')
+
+function findMatchingIndex (sourceItem, target) {
+  var matchingIndex
+  if (Object.keys(sourceItem).includes('name')) {
+    matchingIndex = target
```

Comment:
We can return early here instead of redefining a variable:

```suggestion
    return target
```